src/util/psbt/error.rs

```diff
@@ -76,6 +76,9 @@ pub enum Error {
     MergeConflict(String),
     /// Serialization error in bitcoin consensus-encoded structures
     ConsensusEncoding,
+    #[cfg(feature = "base64")]
```

Comment:
I think @apoelstra's [objection from #557 still applies](https://github.com/rust-bitcoin/rust-bitcoin/pull/557#discussion_r573906238).

Reply:
Not sure I understand it. This PR is intended for a major version release

Reply:
> You cannot have a feature-gated error variant, in a major release **or otherwise**.

Emphasis mine. I explained my understanding of the subtle problem further down in the thread (I wasn't aware of it either):

> Let's imagine crate A depends on rust-bitcoin without features and [also depends on] crate B which also depends on rust-bitcoin without features, but doesn't expose it externally. Now B decides to turn on base64 and does so in a minor release since it doesn't break the API (rust-bitcoin is a purely internal dependency). Now what cargo does is, when compiling one rust-bitcoin for both A and B, it calculates the union of all activated features in the dependency tree and uses that, breaking any previously exhaustive matches in A.

Reply:
Features are additive so that's right, with that in mind I would recommend adding an special error type for FromStr implementation like rust `std` crate does, like `ParseXxxError`. In this case:

```rust
pub enum ParsePsbtError {
    Pbst(super::Error),
    Base64(base64:DecodeError),
}
```

The other option is using `#[non_exahustive]` on the `Error` enum but severely limits that type.

Reply:
@jeandudey neither of this will work: there is no `#[non_exhaustive]` in rust 1.29 and even with separate error type you will no be able to build if `base64` feature is not used (since there will be no `base64:DecodeError` and you will get a compile-time error).

@sgeisler thank you for the explanation, I see the problem now... Well, the only thing which can be done in this respect is to keep error as a string value...

Reply:
>you will no be able to build if base64 feature is not used (since there will be no base64:DecodeError and you will get a compile-time error).

That type should be feature gated too, the other people's code  that depen on the base64 feature will use it when it's enabled and other users will still be able to use the other error type without problems.

So a `#[cfg(feature = "base64")] pub enum ParsePsbtError` would still work

Reply:
The point made is that to use get `ParsePsbtError` you have to enable it (and acts as an "addition") instead of getting it directly on the `Error` enum when other crate enables it.

Reply:
Oh, I see.. But that arguably even worse than just having a string embedded into a error...

Reply:
Stringly typed errors are pretty useless in library code :disappointed: Just imagine wanting to display a meaningful error message in another language. So @jeandudey's solution sounds like the least bad one to me.

Reply:
Ok, did it: https://github.com/rust-bitcoin/rust-bitcoin/pull/557/commits/fc006bb96fb5ad40fb70222b82cb141429a977b8

Since it is a non-API-breaking change put it in the original PR: https://github.com/rust-bitcoin/rust-bitcoin/pull/557

With it, this PR is not needed anymore